src/main/java/com/faforever/api/security/FafServiceAuthenticationToken.java

```diff
@@ -0,0 +1,38 @@
+package com.faforever.api.security;
+
+import com.google.common.collect.ImmutableList;
+import lombok.Getter;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+
+@Getter
+public final class FafServiceAuthenticationToken extends FafAuthenticationToken {
+
+  private final String serviceName;
+
+  public FafServiceAuthenticationToken(
+    String serviceName,
+    @NotNull Collection<FafScope> scopes
+  ) {
+    super(
+      scopes,
+      ImmutableList.<FafRole>builder()
+        .add(new FafRole("SERVICE"))
+        .build()
+    );
+    this.serviceName = serviceName;
+    // since the access token was already verified, each FafAuthenticationToken is implicitly authenticated
+    this.setAuthenticated(true);
```

Comment:
Is this needed since it is set in the super constructor?

Reply:
nah, copy pasta noob